src/components/AddItemForm.js

```diff
@@ -1,28 +1,75 @@
 import React from 'react';
-import { collection, addDoc } from 'firebase/firestore';
+import {
+  collection,
+  addDoc,
+  query,
+  where,
+  onSnapshot,
+  getDocs,
+} from 'firebase/firestore';
 import { db } from '../lib/firebase';
 import './AddItemForm.css';
 
 function AddItemForm() {
-  const submitItem = (event) => {
+  const normalizeValue = (value) => {
+    const punctuationRegex = /[^\w]/gi;
```

Comment:
`\w` matches for words & *underscore*. Which resulted in various forms of `ice cream` (🍦 ice cream, ice cream, ice_cream). I'm also not sure why I could add `ice cream` again since you already have a version of `ice cream` with an emoji version. 

<img width="446" alt="Screen Shot 2021-10-29 at 2 34 35 PM" src="https://user-images.githubusercontent.com/3861088/139504520-4d9281c0-8933-434d-aa71-48172cb880fb.png">

Maybe this matcher would help? `[\W_]`

Reply:
@jonathanyeong Thanks for catching that! Will make the changes to the regex. We missed testing it out with the underscore. For the `ice cream` without the emoji you were able to add, it was because the `ice cream` with an emoji was missing the `itemNameNormalized` field.  I will clean up the database and remove the documents without the `itemNameNormalized` field for the `dandy demon salvo` token.

Reply:
Ahh gotcha! That makes sense!